qcengine/programs/qchem.py

```diff
@@ -259,3 +263,371 @@ def parse_output(self, outfiles: Dict[str, str], input_model: "AtomicInput") ->
+            "n12",
+            "olyp",
+            "pbeop",
+            "pbesol",
+            "pw91",
+            "rpbe",
+            "rvv10",
+            "sogga",
+            "sogga11",
+            "vv10",
+            "b97m-v",
+            "b97m-rv",
+            "m06-l",
+            "tpss",
+            "revtpss",
+            "bloc",
+            "m11-l",
+            "mbeef",
+            "mgga_ms0",
+            "mgga_ms1",
+            "mgga_ms2",
+            "mgga_mvs",
+            "mn12-l",
+            "mn15-l",
+            "otpss",
+            "pkzb",
+            "scan",
+            "t-hcth",
+            "tm",
+            "vsxc",
+            "b3lyp",
+            "pbe0",
+            "revpbe0",
+            "b97",
+            "b1lyp",
+            "b1pw91",
+            "b3lyp5",
+            "b3p86",
+            "b1lyp",
+            "b1pw91",
+            "b3lyp5",
+            "b3p86",
+            "b3pw91",
+            "b5050lyp",
+            "b97-1",
+            "b97-2",
+            "b97-3",
+            "b97-k",
+            "bhhlyp",
+            "hflyp",
+            "mpw1k",
+            "mpw1lyp",
+            "mpw1pbe",
+            "mpw1pw91",
+            "o3lyp",
+            "pbeh-3c",
+            "pbe50",
+            "sogga11-x",
+            "wc04",
+            "wp04",
+            "x3lyp",
+            "m06-2x",
+            "m08-hx",
+            "tpssh",
+            "revtpssh",
+            "b1b95",
+            "b3tlap",
+            "bb1k",
+            "bmk",
+            "dldf",
+            "m05",
+            "m05-2x",
+            "m06",
+            "m06-hf",
+            "m08-so",
+            "mgga_ms2h",
+            "mgga_mvsh",
+            "mn15",
+            "mpw1b95",
+            "mpwb1k",
+            "pw6b95",
+            "pwb6k",
+            "scan0",
+            "t-hcthh",
+            "tpss0",
+            "wb97x-v",
+            "wb97x-d3",
+            "wb97x-d",
+            "cam-b3lyp",
+            "cam-qtp00",
+            "cam-qtp01",
+            "hse-hjs",
+            "lc-rvv10",
+            "lc-vv10",
+            "lc-wpbe08",
+            "lrc-bop",
+            "lrc-wpbe",
+            "lrc-wpbeh",
+            "n12-sx",
+            "rcam-b3lyp",
+            "wb97",
+            "wb97x",
+            "wb97x-rv",
+            "wb97m-v",
+            "m11",
+            "mn12-sx",
+            "wb97m-rv",
+            "wm05-d",
+            "wm06-d3",
+            "dsd-pbepbe-d3",
+            "wb97x-2(lp)",
+            "wb97x-2(tqz)",
+            "xyg3",
+            "xygj-os",
+            "b2plyp",
+            "b2gpplyp",
+            "dsd-pbep86-d3",
+            "ls1dh-pbe",
+            "pbe-qidh",
+            "pbe0-2",
+            "pbe0-dh",
+            "ptpss-d3",
+            "dsd-pbeb95-d3",
+            "pwpb95-d3",
+        }
+        _mp2_methods = {"rimp2"}
+
+        method = input_dict["model"]["method"].lower()
```

Comment:
Rough strategy here to have every method listed out, unsure if there is a better way. @loriab?

Reply:
I'm hesitant to use this approach because it requires knowledge of method hierarchy. This is obvious when different methods have different costs, but will have a lot of edge cases such as SCS-MP2, solvation, excited states, etc.

The methods list comes from Q-Chem's manual, so I see it as being on them if one of these does not work.

Reply:
I completely agree that if any method doesn't run to "Have a nice day", it's on them. It's whether the final energy result for all is to be found at "SCF energy in the final basis set" that makes me nervous for methods as different as b3lyp, pbeh-3c, double hybrids, vv10's, etc.

Reply:
Fair point that's a bug; it should be "Total energy in the final basis set". (If Q-Chem fails to put the correct result for an SCF method in "Total energy in the final basis set", then I'm willing to say that's on them.) 

This does raise a question about whether the `scf_total_energy` should be  "Total energy in the final basis set" or  "SCF energy in the final basis set"...

Reply:
In terms of schema?

Reply:
Yes.

Reply:
Can you explain a bit more here? Why are basis sets entering this conversation when a basis is pinned by the input?

Reply:
"basis" is just in the line that's scraped into `scf_total_energy`, which in turn for this whole list becomes the `return_energy`. I'm gloomier than Matt that all this list of methods is so consistently scraped with one line. unless I've lost track of the real issue ...